src/app/ui/palette_listbox.cpp

```diff
@@ -135,90 +141,100 @@ class PaletteListBox::LoadingItem : public ListItem {
   int m_state;
 };
 
-PaletteListBox::PaletteListBox()
-  : m_resourcesLoader(new ResourcesLoader(new PalettesLoaderDelegate))
-  , m_resourcesTimer(100)
-  , m_loadingItem(NULL)
-{
-  m_resourcesTimer.Tick.connect(base::Bind<void>(&PaletteListBox::onTick, this));
-}
-
-doc::Palette* PaletteListBox::selectedPalette()
-{
+PaletteResource* PaletteListBox::selectedPaletteResource() {
   if (PaletteListItem* listItem = dynamic_cast<PaletteListItem*>(getSelectedChild()))
-    return listItem->paletteResource()->palette();
+    return listItem->paletteResource();
   else
-    return NULL;
+    return nullptr;
 }
 
-bool PaletteListBox::onProcessMessage(ui::Message* msg)
-{
-  switch (msg->type()) {
-
-    case kOpenMessage: {
-      m_resourcesTimer.start();
-      break;
-    }
+doc::Palette* PaletteListBox::selectedPalette() {
+  auto resource = selectedPaletteResource();
+  return resource ? resource->palette() : nullptr;
+}
 
-  }
-  return ListBox::onProcessMessage(msg);
+std::string PaletteListBox::selectedPaletteName() {
+  auto resource = selectedPaletteResource();
+  return resource ? resource->name() : "";
 }
 
-void PaletteListBox::onChange()
-{
-  doc::Palette* palette = selectedPalette();
+void PaletteListBox::onChange() {
+  auto palette = selectedPalette();
   if (palette)
     PalChange(palette);
```

Comment:
more of a not for the future: `PalChange` is kind of bad as a name

Reply:
Agreed. I assume it's like this because Change already exists in the base class.
Changing it would involve redirecting all the places that used it and I wanted to limit the PR from growing too much.